File: csharp/demo/demo12b01-data-race-single.cs
/*
 * DATA RACES
 * Version 01: Without multithreading
 */
using System;
using System.Threading;
using System.Linq;



class Demo12B01 : IRunnable
{
    public void run()
    {
        const int N = 8;
        int result = getResult(N);
        Console.WriteLine("Numbers of integers that are divisible by 2 or 3 is: " + result);
    }


    private int getResult(int N)
    {
        var a = Enumerable.Repeat(false, N + 1).ToArray();

        for (int i = 1; i <= N; ++i)
            if (i % 2 == 0 || i % 3 == 0)
                a[i] = true;

        // res = number of true values in array
        int res = a.Count(val => val);
        return res;
    }
}

File: python/demo12b01_data_race_single.py
'''
DATA RACES
Version 01: Without multithreading
'''



def get_result(n: int):
    a = [False] * (n + 1)

    for i in range(1, n + 1):
        if i % 2 == 0 or i % 3 == 0:
            a[i] = True

    res = a.count(True)
    return res



N = 8
result = get_result(N)
print('Number of integers that are divisible by 2 or 3 is:', result)
